pulp_smash/tests/puppet/api_v2/test_sync_publish.py

```diff
@@ -0,0 +1,655 @@
+        ))
+
+    def test_id_notes(self):
+        """Validate the ``id`` and ``notes`` attributes for each repo."""
+        for key in ('id', 'notes'):
+            for body, attrs in zip(self.bodies, self.attrs_iter):
+                with self.subTest((key, body, attrs)):
+                    self.assertIn(key, attrs)
+                    self.assertEqual(body[key], attrs[key])
+
+    def test_number_importers(self):
+        """Each repository should have only one importer."""
+        for i, importers in enumerate(self.importers_iter):
+            with self.subTest(i=i):
+                self.assertEqual(len(importers), 1, importers)
+
+    def test_importer_type_id(self):
+        """Validate the ``importer_type_id`` attribute of each importer."""
+        key = 'importer_type_id'
+        for body, importers in zip(self.bodies, self.importers_iter):
+            with self.subTest((body, importers)):
+                self.assertIn(key, importers[0])
+                self.assertEqual(body[key], importers[0][key])
+
+    def test_importer_config(self):
+        """Validate the ``config`` attribute of each importer."""
+        key = 'config'
+        for body, importers in zip(self.bodies, self.importers_iter):
+            with self.subTest((body, importers)):
+                self.assertIn(key, importers[0])
+                self.assertEqual(body['importer_' + key], importers[0][key])
+
+
+class SyncValidFeedTestCase(_BaseTestCase):
+    """Test creating repository with valid feed.
+
+    Create two repositories wih valid feeds, one with valid and one with
+    invalid query. Check that sync finish succesfully in both of them.
+
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Create an puppet repositories with a valid feed and sync it."""
+        super(SyncValidFeedTestCase, cls).setUpClass()
+        bodies = tuple((_gen_puppet_repo_body() for i in range(2)))
+        bodies[0]['importer_config'] = {
+            'feed': _VALID_PUPPET_FEED,
+            'queries': [_VALID_PUPPET_QUERY],
+        }
+        bodies[1]['importer_config'] = {
+            'feed': _VALID_PUPPET_FEED,
+            'queries': [_INVALID_PUPPET_QUERY],
+        }
+        cls.attrs_iter = tuple((
+            create_repository(cls.cfg, body) for body in bodies
+        ))
+        cls.sync_repo = []  # raw responses
+        cls.task_bodies = tuple((chain.from_iterable(  # response bodies
+            poll_spawned_tasks(cls.cfg, call_report)
+            for call_report in (_sync_repo(
+                cls.cfg,
+                attr_iter['_href'],
+                cls.sync_repo
+            ) for attr_iter in cls.attrs_iter)
+        )))
+
+    def test_start_sync_code(self):
+        """Assert the call to sync each repository returns an HTTP 202."""
+        for i in range(2):
+            with self.subTest(i=i):
+                self.assertEqual(self.sync_repo[i].status_code, 202)
```

Comment:
Please avoid this hard-coded range and, instead, iterate directly through the objects being tested. The `i` variable is only generated for ease of troubleshooting should the test fail.

```python
        for i, response in enumerate(self.sync_repo):
            with self.subTest(i=i):
                self.assertEqual(resopnse.status_code, 202)
```